bootstrap/handlers/httpserver.go

```diff
@@ -156,7 +168,70 @@ func (b *HttpServer) BootstrapHandler(
 		}()
 
 		b.isRunning = true
-		err := server.ListenAndServe()
+
+		var ln net.Listener
+		switch bootstrapConfig.Service.SecurityOptions["Mode"] {
+		case "zerotrust":
+			secretProvider := container.SecretProviderExtFrom(dic.Get)
+			var zitiCtx ziti.Context
+			var ctxErr error
+			if secretProvider != nil {
```

Comment:
secretProvider  should never be nil. It will either be the `SecureProvider` or the `InsecureProvider`

If it is nil, then log error and return false.

Reply:
instead, i'll set `err` and let it continue down to line 231 where it checks the err and then either runs the server or exits in the next if block... that seems to be the pattern here. (note that this is within a goroutine and doesn't follow the established pattern of returning false)